tuf/data/root.go

```diff
@@ -23,14 +24,45 @@ type Root struct {
 	ConsistentSnapshot bool                 `json:"consistent_snapshot"`
 }
 
+// isValidRootStructure returns an error, or nil, depending on whether the content of the struct
+// is valid for root metadata.  This does not check signatures or expiry, just that
+// the metadata content is valid.
+func isValidRootStructure(r Root) error {
+	expectedType := TUFTypes[CanonicalRootRole]
+	if r.Type != expectedType {
+		return ErrInvalidMetadata{
+			role: CanonicalRootRole, msg: fmt.Sprintf("expected type %s, not %s", expectedType, r.Type)}
+	}
+	if len(r.Roles) < 4 {
+		return ErrInvalidMetadata{role: CanonicalRootRole, msg: "does not have all required roles"}
+	} else if len(r.Roles) > 4 {
```

Comment:
It's OK to have additional roles, particularly as we may add the mirrors role at some point.